### ISSUE TITLE
Error linke not working

### ISSUE THREAD

**Issue by serendipity1004** [zino-app/graphql-flutter#516]
I am using below code to concat error link to auth link and http link. However, the prints in the ErrorLink do not show when I purposefully cause error. I might be using it wrong because it's not documented. Any suggestions what might be wrong?

```
final HttpLink httpLink = HttpLink(
  uri: CONFIG.domain + '/apollo_api',
);

final CustomAuthLink authLink = CustomAuthLink(getHeaders: () async {
  final String token = await Storage.getAuthToken();
  final packageInfo = await PackageInfo.fromPlatform();

  final String buildNo = packageInfo.buildNumber;
  String os;

  if (Platform.isIOS) {
    os = 'iOS';
  } else if (Platform.isAndroid) {
    os = 'android';
  }

  return {
    'authorization': token,
    'os': os,
    'buildNo': buildNo,
  };
});

final ErrorLink errorLink = ErrorLink(
  errorHandler: (ErrorResponse response) {
    print(response);
    if (response.exception is OperationException) {
      print('exception:${response.exception}');
    }
  },
);

final Link link = authLink.concat(httpLink).concat(errorLink);

client = GraphQLClient(
  link: link,
  cache: NormalizedInMemoryCache(
    dataIdFromObject: cacheKeyBuilder,
  ),
);
```

**Comment by bsr203** [comment]
can you try `authLink.concat(errorLink.concat(httpLink))` . At least this is how I do it

**Comment by serendipity1004** [comment]
@bsr203 unfortunately it does not work. What version of graphql-flutter are you using? I am currently using   `graphql_flutter: ^2.1.1-beta.4`

**Comment by bsr203** [comment]
`  graphql_flutter: ^2.1.1-beta.5`

**Comment by mainawycliffe** [comment]
What about `final Link link = authLink..concat(httpLink)..concat(errorLink);`? 

And is it working when you concat just two Links? You can also try the [Link.from](https://pub.dev/documentation/graphql/2.1.1-beta.5/graphql/Link/from.html) function and see if it works.

**Comment by ekoto4** [comment]
@serendipity1004 @mainawycliffe 
Did anyone find a fix to this? I am also running into this problem. It seems that no matter the order, the linking stops working correctly when the http link is concatenated. Anything concatenated afterwards does not work. Thank you for the help in advance!

**Comment by BaptisteMonet** [comment]
same problem for me, any new or idea ?

**Comment by ekoto4** [comment]
> same problem for me, any news or idea ?

The order the links are concatenated matters. This ow i set it up to get a token that was passed in the header.

```  static final Link _link = _tokenLink.concat(_authLink.concat(_httpLink)); ```

I think you can substitute tokenLink for errorLink and it should work.

**Comment by BaptisteMonet** [comment]
ok thx for your answer but I try with [this](https://github.com/zino-app/graphql-flutter/issues/521#issuecomment-624600089)

and it works

**Comment by micimize** [comment]
This is outdated / likely solved by using `gql_links` in `4.0.0-beta.1` #648. Please open a new issue if not
